third_party/docfx/templates/devsite/ManagedReference.common.js

```diff
@@ -311,4 +319,22 @@ function handleItem(vm, gitContribute, gitUrlPattern) {
       vm.status_title = "Early access"
     }
   }
+
+  if (vm.javaType) {
+    // Resetting 'type' from added field 'javaType' field here 
+    // because docfx cannot process custom types while using ManagedReference.
+    // This allows Java to use custom types without rewriting for UniversalReference
+    if (vm.javaType === "exception") {
+      vm.type = "exception";

```

Comment:
Do we need these `if` conditions? Can we do `vm.type = vm.javaType`?

Reply:
There were some javaTypes that I didn't want type overwritten for, that's why I was including the if statements. But I agree this would be better so I'm going to move this logic to the doclet so that javaType is only included if we want to use it in the template. Then I can do what you said with `vm.type = vm.javaType`.